src/MakingSense.AspNetCore.Documentation/DocumentationMiddleware.cs

```diff
@@ -73,14 +74,17 @@ private void ParseLayout()
 		public async Task Invoke(HttpContext context)
 		{
 			PathString subpath;
+			string lang;
+
 			if (context.Request.IsAGetRequest()
-				&& TryGetDocumentsSubpath(context.Request, out subpath))
+				&& TryGetDocumentsSubpathAndLanguage(context.Request, out subpath, out lang))
 			{
 				bool isDirectory;
 				string filePath = GetFilePath(subpath, out isDirectory);
+				var languages = GetLanguages(lang);
 
 				IDocumentHandler handler;
-				var foundHandler = _documentHandlerResolver.TryResolveHandler(_options.FileProvider, filePath, out handler);
+				var foundHandler = _documentHandlerResolver.TryResolveHandler(_options.FileProvider, filePath, languages, out handler);
```

Comment:
Another option is to get _language_ out of _documentHandler_. We could try to find the right language here.

Something like:
```
var specificLanguageFilePath = $"{filePath}.{lang}";
var foundHandler = _documentHandlerResolver.TryResolveHandler(_options.FileProvider, specificLanguageFilePath, out handler);
    ?? _documentHandlerResolver.TryResolveHandler(_options.FileProvider, filePath, out handler);
```

In that way, we could remove the changes from _DocumentHandlerResolvers_ and _DocumentHandlers_.